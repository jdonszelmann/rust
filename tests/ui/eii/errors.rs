//@ compile-flags: --crate-type rlib
#![feature(eii)]
#![feature(decl_macro)]
#![feature(rustc_attrs)]
#![feature(eii_internals)]

#[core::eii_macro_for(bar)] //~ ERROR `#[eii_macro_for(...)]` is only valid on macros
fn hello() {
    #[core::eii_macro_for(bar)] //~ ERROR `#[eii_macro_for(...)]` is only valid on macros
    let x = 3 + 3;
}

#[core::eii_macro_for] //~ ERROR `#[eii_macro_for(...)]` expects a list of one or two elements
#[core::eii_macro_for()] //~ ERROR `#[eii_macro_for(...)]` expects a list of one or two elements
#[core::eii_macro_for(bar, hello)] //~ ERROR expected this argument to be "unsafe"
#[core::eii_macro_for(bar, "unsafe", hello)] //~ ERROR `#[eii_macro_for(...)]` expects a list of one or two elements
#[core::eii_macro_for(bar, hello, "unsafe")] //~ ERROR `#[eii_macro_for(...)]` expects a list of one or two elements
#[core::eii_macro_for = "unsafe"] //~ ERROR `#[eii_macro_for(...)]` expects a list of one or two elements
#[core::eii_macro_for(bar)]
#[rustc_builtin_macro(eii_macro)]
macro foo() {}

unsafe extern "Rust" {
    safe fn bar(x: u64) -> u64;
}

#[foo] //~ ERROR `#[foo]` is only valid on functions
static X: u64 = 4;
#[foo] //~ ERROR `#[foo]` is only valid on functions
const Y: u64 = 4;
#[foo] //~ ERROR `#[foo]` is only valid on functions
macro bar() {}

#[foo()]
//~^ ERROR `#[foo]` expected no arguments or a single argument: `#[foo(default)]`
#[foo(default, bar)]
//~^ ERROR `#[foo]` expected no arguments or a single argument: `#[foo(default)]`
#[foo("default")]
//~^ ERROR `#[foo]` expected no arguments or a single argument: `#[foo(default)]`
#[foo = "default"]
//~^ ERROR `#[foo]` expected no arguments or a single argument: `#[foo(default)]`
fn other(x: u64) -> u64 {
    x
}
