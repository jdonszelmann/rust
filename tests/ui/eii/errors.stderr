error: `#[eii_macro_for(...)]` is only valid on macros
  --> $DIR/errors.rs:7:1
   |
LL | #[core::eii_macro_for(bar)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `#[eii_macro_for(...)]` is only valid on macros
  --> $DIR/errors.rs:9:5
   |
LL |     #[core::eii_macro_for(bar)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `#[eii_macro_for(...)]` expects a list of one or two elements
  --> $DIR/errors.rs:13:1
   |
LL | #[core::eii_macro_for]
   | ^^^^^^^^^^^^^^^^^^^^^^

error: `#[eii_macro_for(...)]` expects a list of one or two elements
  --> $DIR/errors.rs:14:1
   |
LL | #[core::eii_macro_for()]
   | ^^^^^^^^^^^^^^^^^^^^^^^^

error: expected this argument to be "unsafe".
  --> $DIR/errors.rs:15:28
   |
LL | #[core::eii_macro_for(bar, hello)]
   |                            ^^^^^
   |
note: the second argument is optional
  --> $DIR/errors.rs:15:28
   |
LL | #[core::eii_macro_for(bar, hello)]
   |                            ^^^^^

error: `#[eii_macro_for(...)]` expects a list of one or two elements
  --> $DIR/errors.rs:16:1
   |
LL | #[core::eii_macro_for(bar, "unsafe", hello)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `#[eii_macro_for(...)]` expects a list of one or two elements
  --> $DIR/errors.rs:17:1
   |
LL | #[core::eii_macro_for(bar, hello, "unsafe")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `#[eii_macro_for(...)]` expects a list of one or two elements
  --> $DIR/errors.rs:18:1
   |
LL | #[core::eii_macro_for = "unsafe"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `#[foo]` is only valid on functions
  --> $DIR/errors.rs:27:1
   |
LL | #[foo]
   | ^^^^^^

error: `#[foo]` is only valid on functions
  --> $DIR/errors.rs:29:1
   |
LL | #[foo]
   | ^^^^^^

error: `#[foo]` is only valid on functions
  --> $DIR/errors.rs:31:1
   |
LL | #[foo]
   | ^^^^^^

error: `#[foo]` expected no arguments or a single argument: `#[foo(default)]`
  --> $DIR/errors.rs:34:1
   |
LL | #[foo()]
   | ^^^^^^^^

error: `#[foo]` expected no arguments or a single argument: `#[foo(default)]`
  --> $DIR/errors.rs:36:1
   |
LL | #[foo(default, bar)]
   | ^^^^^^^^^^^^^^^^^^^^

error: `#[foo]` expected no arguments or a single argument: `#[foo(default)]`
  --> $DIR/errors.rs:38:1
   |
LL | #[foo("default")]
   | ^^^^^^^^^^^^^^^^^

error: `#[foo]` expected no arguments or a single argument: `#[foo(default)]`
  --> $DIR/errors.rs:40:1
   |
LL | #[foo = "default"]
   | ^^^^^^^^^^^^^^^^^^

error: aborting due to 15 previous errors

